src/server.cpp

```diff
@@ -206,7 +206,8 @@ Server::Server(
 	infostream<<"- game:   "<<m_gamespec.path<<std::endl;
 
 	// Create world if it doesn't exist
-	if(!loadGameConfAndInitWorld(m_path_world, m_gamespec))
+	std::string name = m_path_world.substr(m_path_world.find_last_of(DIR_DELIM) + 1);
```

Comment:
I'd prefer if you used the appropriate `fs` function for this (there's a function called `fs::RemoveLastPathComponent` or something like that).